include/pnet_api.h

```diff
@@ -1213,10 +1201,10 @@ typedef struct pnet_ip_cfg
  */
 typedef struct pnet_if_cfg
 {
-   pnet_netif_t main_port; /**< Main (DAP) network interface. */
+   const char * main_netif_name; /**< Main (DAP) network interface. */
```

Comment:
In `pf_fspm_init()`, a copy of the configuration will be stored. When the string was allocated as an array in `pnet_cfg_t`, this would automatically copy the string. As the sting is now a pointer, no such copy will be made. This may be a problem if user allocated the string in non-static memory, as the content of the string could then change after the call to `pnet_init()`.

This could be solved by requiring that user allocates the string in static storage, or by making a copy of the string in `pf_fspm_init()`. I would recommend the second option, as it is more user-friendly.

Reply:
Netif names part of p_cfg will be copied into net->pf_interface data structure during initialization. I removed all other use of netif names in p_cfg. For example pf_port_get_state() is used instead of pf_port_get_config().
Done